src/form-engine.test.tsx

```diff
@@ -215,6 +216,33 @@ describe('Form engine component', () => {
     });
   });
 
+  describe('Read only mode', () => {
+    it('should ascertain that each field with readonly = true passed will not be editable', async () => {
+      await act(async () => {
+        renderForm(null, readOnlyValidationForm);
+      });
+
+      const visitTypeDropdown = screen.getByRole('combobox', {
+        name: /visit type/i,
+      });
+      const visitPunctualityTextbox = screen.getByLabelText(/visit punctuality/i);
+
+      expect(visitTypeDropdown).toBeInTheDocument();
+      expect(visitTypeDropdown).toHaveClass('cds--list-box__field');
+
+      const visitTypeWrapper = visitTypeDropdown.closest('.cds--dropdown');
+      expect(visitTypeWrapper).toHaveClass('cds--dropdown cds--dropdown--readonly cds--list-box');
+
+      expect(visitPunctualityTextbox).toBeInTheDocument();
+      expect(visitPunctualityTextbox).toHaveClass('cds--text-input');
+
+      const visitPunctualityWrapper = visitPunctualityTextbox.closest('.cds--text-input-wrapper');
+      expect(visitPunctualityWrapper).toHaveClass(
+        'cds--form-item cds--text-input-wrapper cds--text-input-wrapper--readonly',
+      );
+    });
+  });
```

Comment:
I love tests but this test is supposed to be testing a form rendered in "view" mode. I've updated the ticket to clearly reflect this. (sorry for the confusion)

> The form-engine supports three major modes: enter (new), edit and view (readonly). This ticket is meant for adding test coverage of the form’s behaviour while in view mode.

Reply:
No worries, let me do a rework